File: samples/output/jpa/src/main/javagen/topmodel/jpa/sample/demo/dtos/securite/utilisateur/UtilisateurItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package topmodel.jpa.sample.demo.dtos.securite.utilisateur;

import jakarta.annotation.Generated;

import topmodel.jpa.sample.demo.enums.securite.utilisateur.TypeUtilisateurCode;

/**
 * Détail d'un utilisateur en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface UtilisateurItem {

	/**
	 * Id de l'utilisateur.
	 */
	Integer getId();

	/**
	 * Nom de l'utilisateur.
	 */
	String getNom();

	/**
	 * Nom de l'utilisateur.
	 */
	String getPrenom();

	/**
	 * Email de l'utilisateur.
	 */
	String getEmail();

	/**
	 * Type d'utilisateur.
	 */
	TypeUtilisateurCode getTypeUtilisateurCode();

	/**
	 * Hydrate values of instance.
	 * @param id value to set.
	 * @param nom value to set.
	 * @param prenom value to set.
	 * @param email value to set.
	 * @param typeUtilisateurCode value to set.
	 */
	void hydrate(Integer id, String nom, String prenom, String email, TypeUtilisateurCode typeUtilisateurCode);
}

File: samples/output/focus/src/model/securite/utilisateur/utilisateur-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {e, entity, EntityToType} from "@focus4/entities";
import {DO_CODE, DO_EMAIL, DO_ID, DO_LIBELLE} from "../../../domains";

import {TypeUtilisateurCode} from "./references";

export type UtilisateurItem = EntityToType<UtilisateurItemEntityType>;
export type UtilisateurItemEntityType = typeof UtilisateurItemEntity;

export const UtilisateurItemEntity = entity({
    id: e.field(DO_ID, f => f.optional()
        .label("securite.utilisateur.utilisateur.id")
    ),
    nom: e.field(DO_LIBELLE, f => f
        .label("securite.utilisateur.utilisateur.nom")
    ),
    prenom: e.field(DO_LIBELLE, f => f
        .label("securite.utilisateur.utilisateur.prenom")
    ),
    email: e.field(DO_EMAIL, f => f
        .label("securite.utilisateur.utilisateur.email")
    ),
    typeUtilisateurCode: e.field(DO_CODE, f => f.type<TypeUtilisateurCode>().defaultValue("GEST")
        .label("securite.utilisateur.utilisateur.typeUtilisateurCode")
    )
});
